src/Corax/Utils/GrowableBitArray.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Runtime.CompilerServices;
+using Sparrow.Server;
+
+namespace Corax.Utils;
+
+internal unsafe struct GrowableBitArray : IDisposable
+{
+    internal static readonly int MaxCapacityPerBitmap = (int.MaxValue - sizeof(ByteStringStorage)) / sizeof(ulong);
+    internal static readonly long MaxCapacityPerBitmapInBits = MaxCapacityPerBitmap * 64L;
+    private BitArray[] _bitArrays;
+    private readonly long _capacity;
+    
+    /// <summary>
+    /// Creates a new growable bit array. It accepts when bits id between [0, capacity]
+    /// </summary>
+
+    public GrowableBitArray(ByteStringContext allocator, long capacity)
+    {
+        _capacity = capacity + 1; // ensure it's not zero and handles the last bit inclusively.
+        
+        var numberOfUlongsToAllocate = _capacity / 64 + (_capacity % 64 == 0 ? 0 : 1);
+        var numberOfBitArrays = (int)Math.Ceiling(numberOfUlongsToAllocate / (double)MaxCapacityPerBitmap);
+        _bitArrays = new BitArray[numberOfBitArrays];
+        var lastChunkSize = (int)(numberOfUlongsToAllocate - (long)(numberOfBitArrays - 1) * MaxCapacityPerBitmap);
+        for (int i = 0; i < numberOfBitArrays; ++i)
+        {
+            _bitArrays[i] = new BitArray(allocator, i == numberOfBitArrays - 1
+                ? lastChunkSize
+                : MaxCapacityPerBitmap);
+
+        }
+    }
+
+#if DEBUG
+    public bool IsValid
+    {
+        get
+        {
+            for (int i = 0; i < _bitArrays.Length; ++i)
+                if (_bitArrays[i].IsValid == false)
+                    return false;
+            return true;
+        }
+    }
+#endif
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool Add(long id)
+    {
+        if (id > _capacity)
```

Comment:
Shouldn't be `>=`?

Reply:
Bit Arrays as we built them originally are 0 based (you could address the 0-bit). So yes `==` will be over capacity, great catch. I would suggest to add a test to ensure that is the case before fixing it.